GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -0,0 +1,462 @@
+    allocate (omega_weights(num_points))
+
+    do i_point = 1, num_points
+       omega_points(i_point) = x_tw(i_point)
+       omega_weights(i_point) = x_tw(i_point + num_points)
+    end do
+
+    ! scale the minimax parameters
+    omega_points(:) = omega_points(:)*e_min
+    omega_weights(:) = omega_weights(:)*e_min
+
+    ! set up the minimax time grid
+    call get_points_weights_tau(num_points, e_range, x_tw, ierr)
+    if (ierr /= 0) return
+
+    ! For RPA we include already a factor of two (see later steps)
+    scaling = 2.0d0
+
+    allocate (tau_points(num_points))
+    allocate (tau_weights(num_points))
+
+    do i_point = 1, num_points
+       tau_points(i_point) = x_tw(i_point)/scaling
+       tau_weights(i_point) = x_tw(i_point + num_points)/scaling
+    end do
+
+    ! scale grid from [1,R] to [e_min,e_max]
+    tau_points(:) = tau_points(:)/e_min
+    tau_weights(:) = tau_weights(:)/e_min
+
+    allocate (cosft_wt(num_points, num_points))
+    allocate (cosft_tw(num_points, num_points))
+    allocate (sinft_wt(num_points, num_points))
+
+    ! get the weights for the cosine transform W^c(it) -> W^c(iw)
+    call get_transformation_weights(num_points, tau_points, omega_points, cosft_wt, e_min, e_max, &
+         max_errors(1), cos_t_to_cos_w, ierr)
+    if (ierr /= 0) return
+
+    ! get the weights for the cosine transform W^c(iw) -> W^c(it)
+    call get_transformation_weights(num_points, tau_points, omega_points, cosft_tw, e_min, e_max, &
+         max_errors(2), cos_w_to_cos_t, ierr)
+    if (ierr /= 0) return
+
+    ! get the weights for the sine transform Sigma^sin(it) -> Sigma^sin(iw) (PRB 94, 165109 (2016), Eq. 71)
+    call get_transformation_weights(num_points, tau_points, omega_points, sinft_wt, e_min, e_max, &
+         max_errors(3), sin_t_to_sin_w, ierr)
+    if (ierr /= 0) return
+
+    ! Compute the actual weights used for the inhomogeneous cosine/ FT and check whether
+    ! the two matrices for the forward/backward transform are the inverse of each other.
+    do i_point = 1, num_points
+       do j_point = 1, num_points
+          cosft_wt(j_point, i_point) = cosft_wt(j_point, i_point)*cos(tau_points(i_point)*omega_points(j_point))
+          cosft_tw(i_point, j_point) = cosft_tw(i_point, j_point)*cos(tau_points(i_point)*omega_points(j_point))
+          sinft_wt(j_point, i_point) = sinft_wt(j_point, i_point)*sin(tau_points(i_point)*omega_points(j_point))
+       end do
+    end do
+
+    allocate (mat(num_points, num_points))
+    mat(:, :) = matmul(cosft_wt, cosft_tw)
+    do i_point = 1, num_points
+       mat(i_point, i_point) = mat(i_point, i_point) - 1.0d0
+    end do
+    cosft_duality_error = maxval(abs(mat))
+
+    deallocate (mat)
+    deallocate (x_tw)
+
+  end subroutine gx_minimax_grid
+
+  ! **************************************************************************************************
+  ! /brief get the weights eiter for the cosine/sin transformation for tau to omega or viceversa
```

Comment:
General comment for the description headers: does this still follow the doxygen convention? in cp2k it is \brief not /brief and !>, not ! o etc. 
And in some files, like minimax_utils.F90 the headers look like the original cp2k ones

Reply:
You are right. I have modified it following the conventions here: https://www.doxygen.nl/manual/commands.html

I propose to use backslash for the `\brief` command and at for the parameters `@param`